common/environment/setup/install.sh

```diff
@@ -258,3 +258,34 @@ _vmkdir() {
 		install -dm${mode} ${_destdir}/${dir}
 	fi
 }
+
+_vcompletion() {
+	local cmd="$1" shell="$2" file="$3"
+	local _bash_completion_dir=usr/share/bash-completion/completions/
+	local _fish_completion_dir=usr/share/fish/vendor_completions.d/
+	local _zsh_completion_dir=usr/share/zsh/site-functions/
+
+	if [ -z "$DESTDIR" ]; then
+		msg_red "$pkgver: vcompletion: DESTDIR unset, can't continue...\n"
+		return 1
+	fi
+
+	if [ -z "$cmd" -o -z "$shell" ]; then
+		msg_red "$pkgver: vcompletion: command or shell argument unset.\n"
+		return 1
+	fi
+
+	: "${file:=${cmd}.${shell}}"
+	if ! [ -f "$file" ]; then
+		msg_red "$pkgver: vcompletion: file $file doesn't exist.\n"
+	fi
+
+	case "$shell" in
```

Comment:
Other values of `shell` should be error.

Reply:
Definitely, thanks!